Sources/VLCActionSheet.swift

```diff
@@ -0,0 +1,422 @@
+    }
+
+    override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
+        return true
+    }
+}
+
+// MARK: VLCActionSheetCell
+class VLCActionSheetCell: UICollectionViewCell {
+
+    static let identifier = "VLCActionSheetCell"
+
+    let icon: UIImageView = {
+        let icon = UIImageView()
+        icon.translatesAutoresizingMaskIntoConstraints = false
+        icon.contentMode = .scaleAspectFit
+        return icon
+    }()
+
+    let name: UILabel = {
+        let name = UILabel()
+        name.font = UIFont.systemFont(ofSize: 15)
+        name.translatesAutoresizingMaskIntoConstraints = false
+        return name
+    }()
+
+    let stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.spacing = 15.0
+        stackView.axis = .horizontal
+        stackView.alignment = .center
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupViews()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupViews()
+    }
+
+    private func setupViews() {
+        stackView.addArrangedSubview(icon)
+        stackView.addArrangedSubview(name)
+        addSubview(stackView)
+
+        var guide: LayoutAnchorContainer = self
+
+        if #available(iOS 11.0, *) {
+            guide = safeAreaLayoutGuide
+        }
+        NSLayoutConstraint.activate([
+            icon.heightAnchor.constraint(equalToConstant: 25),
+            icon.widthAnchor.constraint(equalTo: icon.heightAnchor),
+
+            name.trailingAnchor.constraint(equalTo: stackView.trailingAnchor),
+            name.centerYAnchor.constraint(equalTo: stackView.centerYAnchor),
+
+            stackView.leadingAnchor.constraint(equalTo: guide.leadingAnchor, constant: 20),
+            stackView.trailingAnchor.constraint(equalTo: guide.trailingAnchor, constant: -20),
+            stackView.heightAnchor.constraint(equalTo: heightAnchor),
+            stackView.topAnchor.constraint(equalTo: topAnchor)
+            ])
+    }
+}
+
+// MARK: VLCActionSheetSectionHeader
+class VLCActionSheetSectionHeader: UIView {
+
+    static let identifier = "VLCActionSheetSectionHeader"
+
+    let title: UILabel = {
+        let title = UILabel()
+        title.font = UIFont.systemFont(ofSize: 16)
+        title.translatesAutoresizingMaskIntoConstraints = false
+        return title
+    }()
+
+    let separator: UIView = {
+        let separator = UIView()
+        separator.backgroundColor = .lightGray
+        separator.translatesAutoresizingMaskIntoConstraints = false
+        return separator
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+        setupSeparator()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupView()
```

Comment:
if you have duplication in the inits use a common init also do you need both inits ? should one of them not be marked as not used?